virtualizarr/readers/safetensors.py

```diff
@@ -0,0 +1,641 @@
+
+        Returns
+        -------
+        ChunkManifest
+            A ChunkManifest object containing a single chunk entry that references
+            the tensor's location within the file.
+
+        Notes
+        -----
+        - Each tensor is represented as a single chunk, regardless of size
+        - Chunk keys are generated as: "0" for 1D, "0.0" for 2D, "0.0.0" for 3D, etc.
+
+        Examples
+        --------
+        For a 3D tensor (shape=(10, 20, 30)) starting at byte 1024 with length 24000:
+        >>> manifest = _create_chunk_manifest("model.safetensors", 1024, 24000, (10, 20, 30))
+        >>> print(manifest.entries.keys())
+        dict_keys(['0.0.0'])
+        """
+        # Create a single chunk key (e.g., "0" for a 1D tensor, "0.0" for a 2D tensor)
+        key_parts = ["0"] * (len(shape) or 1)
+        chunk_key = ChunkKey(".".join(key_parts))
+
+        chunk_entry = ChunkEntry.with_validation(  # type: ignore[attr-defined]
+            path=filepath,
+            offset=offset,
+            length=length,
+        )
+
+        chunk_entries = {chunk_key: chunk_entry}
+
+        return ChunkManifest(entries=chunk_entries)
+
+    @staticmethod
+    def _map_dtype(dtype_str: str) -> np.dtype:
+        """
+        Map SafeTensors dtype string to NumPy dtype.
+
+        SafeTensors uses its own dtype naming convention that needs to be mapped
+        to NumPy dtypes for use in xarray. This method performs that mapping,
+        supporting all standard dtypes including ML-specific types like BF16.
+
+        Parameters
+        ----------
+        dtype_str : str
+            SafeTensors dtype string. Valid values are:
+            - Integer types: "I8", "I16", "I32", "I64"
+            - Unsigned types: "U8", "U16", "U32", "U64"
+            - Float types: "F16", "F32", "F64"
+            - ML types: "BF16" (bfloat16), "F8_E5M2", "F8_E4M3" (float8 variants)
+            - Boolean: "BOOL"
+
+        Returns
+        -------
+        np.dtype
+            Corresponding NumPy dtype object that can be used with xarray.
+
+        Raises
+        ------
+        ValueError
+            If the provided dtype_str is not recognized or supported.
+
+        Notes
+        -----
+        - BF16, F8_E5M2, and F8_E4M3 require the ml_dtypes package
+
+        Examples
+        --------
+        >>> dtype = _map_dtype("F32")
+        >>> print(dtype)
+        float32
+
+        >>> dtype = _map_dtype("BOOL")
+        >>> print(dtype)
+        bool
+
+        >>> dtype = _map_dtype("BF16")
+        >>> print(dtype)
+        bfloat16
+        """
+        try:
+            # this import will register new numpy dtypes as a side-effect e.g. bfloat16
+            import ml_dtypes  # noqa: F401
+        except ImportError:
+            raise ImportError("The ml_dtypes package is required to read safetensors files. Please install it with pip install virtualizarr[safetensors].")
+            
+        dtype_map = {
```

Comment:
Maybe just make this a global variable at the top of the file for visibility?

Reply:
I'm going to leave it as is for now until the Zarr dtype PR is released, and then will chat with you about how best to implement this function.